### PR TITLE
[Feature-2582][Connector] Add Hashes && Lists in Redis Sink

### LINKED ISSUE #2582: [Feature][Connector] Add Hashes && Lists in Redis Sink
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Above

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by shangeyao**
## Purpose of this pull request

close #2582

### CODE REVIEW

**Review by hailin0 — COMMENTED** [review]
Can you add  e2e-testcase & update docs?

**Comment by shangeyao** [comment]
> Can you add e2e-testcase & update docs?

I will complete it.

**Comment by hailin0** [comment]
> Can you add e2e-testcase & update docs?


Testcase reference #2429

**Comment by shangeyao** [comment]
> > Can you add e2e-testcase & update docs?
> 
> Testcase reference #2429

I tried it, please review again.

File: seatunnel-e2e/seatunnel-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/redis/FakeSourceToRedisIT.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Container;
+
+import java.io.IOException;
+
+public class FakeSourceToRedisIT extends SparkContainer {
+
+    @Test
+    public void testFakeSourceToRedisSink() throws IOException, InterruptedException {
+        Container.ExecResult execResult = executeSeaTunnelSparkJob("/jdbc/fakesource_to_redis.conf");
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/redis/FakeSourceToRedisIT.java:70]: reference
https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-e2e/seatunnel-spark-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/redis/RedisIT.java#L51

start redis-server in testcase?

File: seatunnel-e2e/seatunnel-spark-e2e/src/test/resources/redis/fakesource_to_redis.conf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+transform {
+  # If you would like to get more information about how to configure seatunnel and see full list of transform plugins,
+  # please go to https://seatunnel.apache.org/docs/flink/configuration/transform-plugins/Sql
+}
+
+sink {
+  redis {
+      host = "localhost"
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-e2e/src/test/resources/redis/fakesource_to_redis.conf:53]: Connect to redis-server you started in testcase?

**Review by wuchunfu — COMMENTED** [review]
@shangeyao Please fix codestyle problem, thx

File: seatunnel-e2e/seatunnel-spark-e2e/src/test/resources/redis/fakesource_to_redis.conf (hunk 2 above)
Comment (hailin0) [reply seatunnel-e2e/seatunnel-spark-e2e/src/test/resources/redis/fakesource_to_redis.conf:53]:
```suggestion
      host = "spark_e2e_redis"
```

Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-e2e/src/test/resources/redis/fakesource_to_redis.conf:53]:
```suggestion
      host = "spark_e2e_redis"
```

**Comment by EricJoy2048** [comment]
Since this pr has not been updated for too long, I will close it. If necessary, please reopen it.

**Pull request closed without merging**